streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/PassportController.java

```diff
@@ -102,6 +109,46 @@ public RestResponse signin(
         return new RestResponse().data(userInfo);
     }
 
+    @PostMapping("signinbycasdoor")
+    public RestResponse signinByCasdoor(@NotBlank(message = "{required}") String code,
+                                        @NotBlank(message = "{required}") String state) throws Exception {
+        String token = "";
```

Comment:
It is not recommended to throw Exception directly, but rather to handle them on a case-by-case basis.

Reply:
i see this.userService.updateLoginTime(username) and authenticator.authenticate(username, password)  in sigin method also throws Exception so that i think  this.userService.updateLoginTime(username) and          userService.createUser(user2) throws Exception is ok

Reply:
Am I right?

Reply:
Those are probably old codes left over, would you mind to change them together?

Reply:
I change my code , if it is right , i will change signin method toghter

Reply:
@jakiuncle Sorry, I ignored the transaction handling, let's revert to the original, and we'll discuss using unified business code to handle exceptions later, feel free to join the discussion